dbt_subprojects/daily_spellbook/models/polymarket/polygon/polymarket_polygon_capital_actions.sql

```diff
@@ -0,0 +1,61 @@
+{{
+  config(
+    schema = 'polymarket_polygon',
+    alias = 'capital_actions',
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_time','evt_index','tx_hash'],
+    incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')],
+    post_hook = '{{ expose_spells(blockchains = \'["polygon"]\',
+                                  spell_type = "project",
+                                  spell_name = "polymarket",
+                                  contributors = \'["tomfutago"]\') }}'
+  )
+}}
+
+select
+  block_time,
+  block_number,
+  'deposit' as action,
+  "from" as from_address,
+  to as to_address,
+  symbol,
+  amount_raw,
+  amount,
+  amount_usd,
+  evt_index,
+  tx_hash
+from {{ source('tokens_polygon', 'transfers')}}
+where contract_address = 0x2791Bca1f2de4661ED88A30C99A7a9449Aa84174 -- USDC
+  and to in (
+    0x4bfb41d5b3570defd03c39a9a4d8de6bd8b8982e, -- CTF Exchange
+    0xc5d563a36ae78145c45a50134d48a1215220f80a  -- NegRisk CTF Exchange
```

Comment:
ok so how about:
- add `safe_proxies` spell - 2 key columns: `owner` and `proxy` - ideas for better names, plmk
- `capital_actions` will link back to the above spell (instead of hardcoded exchange addresses, as I'm doing currently)

> also koryo uses these 2 addresses, idk how to resolve that delta
> 
> 0x3A3BD7bb9528E159577F7C2e685CC81A765002E2
> 0x2791bca1f2de4661ed88a30c99a7a9449aa84174

not sure, what you mean by resolving delta for these addresses (WCOL & USDC contracts respectively)? I think with user in/out flows only USDC applies, no?

Reply:
> what you mean by resolving delta for these addresses 

lmao I fucked up with the USDC contract.

However, I think my point still stands, albeit in a slight variation:

The USDC only gets "locked up" in the conditional token contracts at 
```
 0x4D97DCd97eC945f40cF65F87097ACe5EA0476045
 0x3A3BD7bb9528E159577F7C2e685CC81A765002E2
 ``` 
 
 However, the market contracts might move them there, not the users directly so this will not be helpful while analysing user actions. 
 
 We could approximate Open Interest from these contracts, but not even that is clean, as there can be stale capital in there. that would be bets that are finished, but the winning tickets have not been redeemed yet.

Reply:
> add safe_proxies spell - 2 key columns: owner and proxy - ideas for better names, plmk
> capital_actions will link back to the above spell (instead of hardcoded exchange addresses, as I'm doing currently)

that sgtm

Reply:
> The USDC only gets "locked up" in the conditional token contracts at
> 
> ```
>  0x4D97DCd97eC945f40cF65F87097ACe5EA0476045
>  0x3A3BD7bb9528E159577F7C2e685CC81A765002E2
> ```
> 
> However, the market contracts might move them there, not the users directly so this will not be helpful while analysing user actions.
> 
> We could approximate Open Interest from these contracts, but not even that is clean, as there can be stale capital in there. that would be bets that are finished, but the winning tickets have not been redeemed yet.

I'm still not clear on what needs to happen here. Is the aim to have both user actions and open interest all in one model? That's rather tricky, me thinks. How about:
- keep current `capital_actions` as it is, so it contains deposits to, withdrawals from and transfers in between proxy safes - so all USDC movements regardless of their source and destination within polymarket
- create a separate `open_interest` spell (eg based on: https://dune.com/queries/3343122)

will this approach capture what you have in mind or will something still be missing?